spring-integration-core/src/main/java/org/springframework/integration/endpoint/MessageProducerSupport.java

```diff
@@ -99,9 +117,6 @@ protected void onInit() {
 	protected void doStart() {
 		Assert.state(this.outputChannel != null || StringUtils.hasText(this.outputChannelName),
 				"'outputChannel' or 'outputChannelName' is required");
```

Comment:
M-m-m. Good catch!
Maybe implement `MessageProducerSupport.afterSingletonsInstantiated()` :smile:  ?

Or at least raise an `IllegalStateException` on the matter from the `GraphServer`...